### ISSUE TITLE
Question about topological changes

### ISSUE THREAD

**Issue by JYChen18** [edoRemelli/MeshSDF#4]
We came across a problem with the comparison between MeshSDF and Marching Cube. In our experiments, we found that  Marching Cube and MeshSDF have a similar effect in handling topological changes. To be more specific, in some cases (Experiment1) both of them can do topological changes while in others (Experiment2) both of them fail, which is contradictory to your claim and makes us confused. In the following, I'll give the settings of our experiment.

Experiment 1: bob_and_spot (which is contained in your code)
We found that Marching Cube can also handle the topological change while the other settings (such as loss, model parameters, etc.) are the same as your experiment of 'bob_and_spot'. We use a resolution of 16 due to the heavy computational pressure of Marching Cube implemented in PyTorch. 

Experiment 2:sphere and double sphere
In this experiment, we initialize a solid sphere (with a radius of 1) and we want to dig a small hole (with a radius of 0.5) in the center of it. The process of this experiment is the same as Experiment1. The only difference between this experiment and Experiment1 is that we use Chamfer Distance loss since the hole is invisible. And we found that both of these two methods failed.

In one word, we see no difference between MeshSDF and Marching Cube in handling topological changes, which confuses us. Appreciate your earliest reply.

**Comment by bguillard** [comment]
Hello and sorry for the very late reply.

We could not find a way to directly backpropagate through marching cubes, and relied on the theoretical arguments of [this paper](https://openaccess.thecvf.com/content_cvpr_2018/html/Liao_Deep_Marching_Cubes_CVPR_2018_paper.html) to discard this route. I'd be glad to discuss it if you have found a way to do it!

About experiment 2: I think there is nothing theoretical preventing our approach from capturing this. I would however say this highly depends on the chosen parametrization, and the loss you apply to optimize it. In your case, did you use a trained DeepSDF? what is the size of the latent code? on which data was it trained?
